neverwinter/nwscript/native/scriptcompcore.cpp

```diff
@@ -1528,6 +1528,143 @@ int32_t CScriptCompiler::OutputError(int32_t nError, CExoString *psFileName, int
 	return nError;
 }
 
+
+// Destructively modify a node and all its children to decay it into a single
+// CONSTANT operation, if possible.
+// This function is safe to call multiple times on the same node.
+BOOL CScriptCompiler::ConstantFoldNode(CScriptParseTreeNode *pNode)
+{
+	if (!pNode)
+		return FALSE;
+
+	// Only fold operations that have two operands
+	// TODO: ~0 unary op?
```

Comment:
This might be useful for bitmasks. Maybe ticket it for later if you don't want to do it here.